aoservicelisting.civix.php

```diff
@@ -305,14 +305,26 @@ public static function emailRule($cid) {
     return TRUE;
   }
 
+  /**
+   * Generate a safe Drupal user name for use
+   * @param array $nameDetails
+   */
+  public static function generateDrupalUserName($nameDetails) {
+    $namefields = ['first_name', 'last_name'];
+    foreach($nameFields as $field) {
+      $nameDetails[$field] = preg_replace('/[^a-zA-Z0-9\/]+/', '', strtolower($nameDetails[$field]));
+    }
+    return $nameDetails['first_name'] . '.' . $nameDetails['last_name'] . $nameDetails['id'];
+  }
 
   public static function createUserAccount($cid) {
     // We create the user account for the primary contact.
     if (empty($cid)) {
       return FALSE;
     }
-    $name = CRM_Core_DAO::executeQuery("SELECT LOWER(CONCAT(first_name, '.', last_name, $cid)) AS name, display_name
+    $name = CRM_Core_DAO::executeQuery("SELECT id, first_name, last_name AS name, display_name
```

Comment:
@seamuslee001 please remove the alias of last_name

Reply:
@monishdeb display_name is still used in L342

Reply:
oh ok my bad, yup its fine then to leave display_name